docs/source/software/systemwide/matlab.md

```diff
@@ -61,95 +61,372 @@ Use the following procedure if you need to install a MATLAB toolbox:
+
+Jobs will now default to running on the HPC cluster rather than submitting to your local machine.
+
+To submit jobs to your local machine instead of the HPC cluster, run the following:
+
+```{code-block} matlab
+>> % Get a handle to the local resources
+>> c = parcluster('local');
+```
+
+### Configuring Jobs
+Prior to submitting jobs to the HPC, various parameters can be assigned and adjusted for your job such as partition, email, job walltime, etc.
+
+To get a handle on the cluster and the current configurations please run the following command in the Command Window:
+
+```{code-block} matlab
+>> % Get a handle to the cluster
+>> c = parcluster;
+```
+
+A required field that needs to be set is the memory per CPU core:
+
+```{code-block} matlab
+>> % Specify memory to use, per core (default: 4gb)
+>> c.AdditionalProperties.MemPerCPU = '6gb';
+```
+
+You need to save the changes after modifying the AdditionalProperties:
+
+```{code-block} matlab
+>> c.saveProfile
+```
+
+To view the values of the currently saved configuration, use the following command in the Command Window:
 
-This section provides instructions for submitting a job to the cluster for scaling calculations on an integer factorization sample problem. The complexity of this problem increases with the magnitude of the number, making it computationally intensive. To perform these calculations, we will use the `myParallelAlgorithmFcn.m` MATLAB function. Please note that this section assumes you have already started a MATLAB Parallel Server.
+```{code-block} matlab
+>> % To view current properties
+>> c.AdditionalProperties
+```
+
+If you need to unset a saved parameter, you can do the following in the Command Window:
+
+```{code-block} matlab
+>> % Turn off email notifications 
+>> c.AdditionalProperties.EmailAddress = '';
+>> c.saveProfile
+```
+
+The following are optional fields that can be set and adjusted for the parcluster and include the following items.
+
+Specifying a constaint such as a [CPU constraint](https://rc-docs.northeastern.edu/en/latest/hardware/hardware_overview.html#using-the-constraint-flag):
+
+```{code-block} matlab
+>> % Specify a constraint 
+>> c.AdditionalProperties.Constraint = 'feature-name';
+```
+
+Setting an email to receive updates on the SLURM job:
+
+```{code-block} matlab
+>> % Request email notification of job status
+>> c.AdditionalProperties.EmailAddress = 'user-id@northeastern.edu';
+```
+
+Setting the number of GPUs used and the type of [GPU](https://rc-docs.northeastern.edu/en/latest/hardware/hardware_overview.html#gpu-hardware):
+
+```{code-block} matlab
+>> % Specify number of GPUs (default: 0)
+>> c.AdditionalProperties.GPUsPerNode = 1;
+>> c.AdditionalProperties.GPUCard = 'gpu-card';
+```
+
+```{note}
+You need to make sure the partition you are submitting the job to is the `gpu` or `multigpu` partition of a private partition that has GPUs available.
```

Comment:
Is there a default for this? If I don't set the partition will it send jobs to my current partition?

The current wording makes is sound like it needs to be done, but I think it might be optional with a "IF you need....."

Reply:
Is there a place for --gres to be set as well for gpu?

Reply:
It would just default to short because it just gets submitted as a SLURM job but that is a great point. I will correct the wording. Thank you!!

Reply:
Soft of for the gres statement. MATLAB automagically submits things properly to SLURM (since it is very formulaic probably not too bad) 

c.AdditionalProperties.Partition = 'gpu'
c.AdditionalProperties.GPUsPerNode = 1
c.AdditionalProperties.GPUCard = 'v100-pcie'

>> job = c.batch(@pwd, 1, {}, 'CurrentFolder','.');
Submit arguments: --ntasks=1 --cpus-per-task=1 --ntasks-per-core=1 --mem-per-cpu=4gb --gres=gpu:v100-pcie:1 -p gpu